### PR TITLE
style: Clean up alert rule styles

### PR DESCRIPTION
**Pull request by lynnagara**
Use the standard trash can button
Use components for form elements where possible

### CODE REVIEW

File: src/sentry/static/sentry/app/views/ruleEditor/index.jsx [hunk 1]
```diff
@@ -288,9 +284,9 @@ const RuleEditor = createReactClass({
             </div>
 
             <div className="actions">
-              <button className="btn btn-primary btn-lg" disabled={loading}>
+              <Button priority="primary" disabled={loading}>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/ruleEditor/index.jsx]: Size is left out, intentional?

File: src/sentry/static/sentry/app/views/ruleEditor/ruleNode.jsx [hunk 2]
```diff
@@ -52,8 +52,10 @@ class RuleNode extends React.Component {
           <span ref="html" dangerouslySetInnerHTML={{__html: node.html}} />
         </RuleNodeForm>
         <RuleNodeControls>
-          <a onClick={this.props.onDelete}>
-            <span className="icon-trash" />
+          <a onClick={this.props.onDelete} className="button button-default button-sm">
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/ruleEditor/ruleNode.jsx]: Why no `Button`?

File: src/sentry/static/sentry/app/views/ruleEditor/index.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/ruleEditor/index.jsx]: Kinda, default looks good

File: src/sentry/static/sentry/app/views/ruleEditor/ruleNode.jsx (hunk 2 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/ruleEditor/ruleNode.jsx]: Button gets <enter> behavior which i wanted to avoid

Comment (billyvg) [reply src/sentry/static/sentry/app/views/ruleEditor/ruleNode.jsx]: does adding `type="button"` solve it?

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/ruleEditor/ruleNode.jsx]: Sort of. But I still prefer the behavior of <a> which is harder to trigger. For example you can't tab to a link.

Comment (billyvg) [reply src/sentry/static/sentry/app/views/ruleEditor/ruleNode.jsx]: `tabIndex=-1`?

Comment (lynnagara) [reply src/sentry/static/sentry/app/views/ruleEditor/ruleNode.jsx]: that works!

**Review by billyvg — APPROVED** [review]

**Pull request merged**